Makefile

```diff
@@ -188,6 +188,10 @@ CFLAGS = $(WASM_CFLAGS) --target=$(TARGET_TRIPLE)
 # WebAssembly floating-point match doesn't trap.
 # TODO: Add -fno-signaling-nans when the compiler supports it.
 CFLAGS += -fno-trapping-math
+# We should have proper declarations for everything.
+CFLAGS += \
+    -Werror=missing-declarations \
+    -Werror=implicit-function-declaration \
```

Comment:
Can we instead use `-Werror` and disabled the warngins that are not fixable or not part in code that we maintain?

Reply:
Also maybe this should be its own change.